pkg/resources/provisioning.cattle.io/v1/cluster/mutator_test.go

```diff
@@ -617,3 +618,96 @@ func TestAdmitPreserveUnknownFields(t *testing.T) {
 	assert.Nil(t, response.Patch)
 
 }
+
+func TestDynamicSchemaDrop(t *testing.T) {
```

Comment:
A few test cases that would be good to add:
- Error getting old/new objects from the request (can be hard to mock, don't necessarily need this)
- Annotation is set, but not to true (i.e. annotation is set to false).

Reply:
Added a test case for the latter, AFAIK for the former there should never be an empty `OldObject` on update. I refactored it a bit to do get the old object one level higher.

Reply:
Also refactored the test function to be easier to follow and be more structured per test case.